internal/schedule/scheduler.go

```diff
@@ -22,9 +22,10 @@ type Scheduler struct {
 func (s *Scheduler) Schedule(ctx context.Context, key string, d time.Duration, task func()) error {
 	cancel := make(chan struct{})
```

Comment:
cancel channel is used to make goroutine from line 31 return instead of waiting forever for context.Done() or timer.C 
it would happen if context wasn't cancelled, but task was manually removed -> see Scheduler.Cancel